src/background/hypothesis-chrome-extension.js

```diff
@@ -109,17 +109,20 @@ export default function HypothesisChromeExtension({
   };
 
   function restoreSavedTabState() {
-    store.reload();
-    state.load(store.all());
     chromeTabs.query({}, function (tabs) {
-      tabs.forEach(function (tab) {
-        if (tab.id) {
-          onTabStateChange(tab.id, state.getState(tab.id));
-        }
+      const tabIds = tabs.map(({ id }) => /** @type {number} */ (id));
```

Comment:
Is this cast safe? Can `chrome.tabs.query` return tabs that do not have an ID?

Reply:
I believe it is safe, but to make 100% I will change it in this way:
```suggestion
      const tabIds = tabs.filter(tab => tab.id).map(({ id }) => id));
```